File: templates/python-agent/src/agent_runtime/telemetry.py
from dataclasses import dataclass, field
import re

@dataclass(frozen=True)
class TraceEvent:
    event_type: str
    tenant_id: str
    user_id: str
    session_id: str
    run_id: str
    agent_id: str
    status: str
    data: dict[str, object] = field(default_factory=dict)

def redact(value: str) -> str:
    return re.sub(r"(?i)(token|password|secret|authorization)\s*[:=]\s*\S+", "[REDACTED]", value)

File: templates/typescript-agent/src/telemetry/telemetry.ts
export interface TraceEvent { eventType: string; tenantId: string; userId: string; sessionId: string; runId: string; agentId: string; status: string; data: Record<string, unknown>; }
export interface Telemetry { emit(event: TraceEvent): void; increment(metric: string, value: number, labels: Record<string, string>): void; }
export const redact = (value: string): string => value.replace(/(token|password|secret|authorization)\s*[:=]\s*\S+/gi, "[REDACTED]");
